### PR TITLE
Add minimal IPv6 support to preseed

### PR DESCRIPTION
**Pull request by arjenz**
Fill /etc/network/interfaces with IPv6 stanza if .ip6 is present

### CODE REVIEW

File: provisioning_templates/snippet/preseed_networking_setup.erb [hunk 1]
```diff
@@ -27,6 +27,11 @@ iface $real inet <%= dhcp ? 'dhcp' : 'static' %>
     dns-nameservers <%= @host.subnet.dns_primary %> <%= @host.subnet.dns_secondary %>
     dns-search <%= @host.domain %>
 <% end %>
+<% if @host.ip6 %>
```
Comment (mmoll) [inline provisioning_templates/snippet/preseed_networking_setup.erb]: I think there's also a check for `@host.subnet6` needed here.

File: provisioning_templates/snippet/preseed_networking_setup.erb [hunk 2]
```diff
@@ -47,5 +52,10 @@ iface $real inet <%= dhcp ? 'dhcp' : 'static' %>
     address <%= interface.ip %>
     netmask <%= subnet.mask %>
 <% end %>
+<% if interface.ip6 %>
```
Comment (mmoll) [inline provisioning_templates/snippet/preseed_networking_setup.erb]: I think there's also a check for `interface.subnet6` needed here.

**Review by mmoll — DISMISSED** [review]

**Comment by mmoll** [comment]
I'm a bit unsure about this. I do have IPv6 hosts that got their values filled by facts, but don't have a subnet assigned. Even if they have, `gateway` is not a mandatory parameter in the subnet.

**Comment by arjenz** [comment]
I added a couple of if statements

What would happen was that I'd assign IPv6 through foreman, that would be ignored, and then the node would submit a fact with an autoconfig IPv6 IP (if that happened to be available on the local network), overwriting my assignment.
Without IPv6 autoconfig it will just ignore the IPv6 IP during provisioning, I think I've both seen it clear the assignment in foreman (after a puppet run with no IPv6 on any interface), and foreman sticking to the assignment but nothing acting on it. Not 100% sure on this last behaviour because I was  troubleshooting at the same time.

**Comment by mmoll** [comment]
We're getting there... :)
After another look, I'd like to have a check for the "static" boot-mode to get added and use the added sections only when its set. Then I think all is ready.

**Comment by arjenz** [comment]
And I think the `next if` each loop for the managed_interfaces will skip IPv6 if there is no IPv4 subnet present (because it checks for `interface.subnet.nil?`) currently.
I am not using this part managed_interfaces myself now, added it for completeness sake in this file.

I'll try and fix this and look at the static boot-mode logic tomorrow, thank you for reviewing so far.

**Comment by arjenz** [comment]
@mmoll I reworked the managed_interfaces to handle "IPv6 only" interfaces and added checks for subnet6 existence/boot_modes.

**Comment by mmoll** [comment]
@arjenz Sorry, that slipped off my radar...

I did still have some problems in my environment with the latest changes and started to rework the whole snippet, resulting into a quite big change, also pushed here.

Could you please review and check this in your environment?

**Pull request merged**